l10n_it_vat_registries/wizard/print_registro_iva.py

```diff
@@ -1,52 +1,21 @@
 # -*- coding: utf-8 -*-
-#
-#
-#    Copyright (C) 2011 Associazione OpenERP Italia
-#    (<http://www.openerp-italia.org>).
-#    Copyright (C) 2014-2015 Agile Business Group
-#    (<http://www.agilebg.com>)
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as published
-#    by the Free Software Foundation, either version 3 of the License, or
-#    (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-#
+# Copyright 2011 Associazione OpenERP Italia
+# (<http://www.openerp-italia.org>).
+# Copyright 2014-2017 Lorenzo Battistini - Agile Business Group
+# (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from openerp import models, fields, api, _
-from openerp.exceptions import Warning as UserError
+from odoo import models, fields, api, _
+from odoo.exceptions import Warning as UserError
 
 
 class WizardRegistroIva(models.TransientModel):
-
-    @api.model
-    def _get_period(self):
-        ctx = dict(self._context or {}, account_period_prefer_normal=True)
-        period_ids = self.env[
-            'account.period'].with_context(context=ctx).find()
-        return period_ids
-
     _name = "wizard.registro.iva"
-    _rec_name = "type"
 
-    period_ids = fields.Many2many(
-        'account.period',
-        'registro_iva_periods_rel',
-        'period_id',
-        'registro_id',
-        string='Periods',
-        default=_get_period,
-        help='Select periods you want retrieve documents from',
-        required=True)
-    type = fields.Selection([
+    date_range_id = fields.Many2one('date.range', string="Date range")
```

Comment:
I see that the model 'date.range' is defined by https://github.com/OCA/account-financial-reporting/blob/10.0/account_tax_balance/__manifest__.py#L18 Wouldn't be better add date_range to depends explicitly?